concepts/05 Widgets/FileManager/10 Bind to File Systems/30 Custom File System.md

```diff
@@ -2,59 +2,205 @@ Use the **FileManager** widget's [Custom](/api-reference/10%20UI%20Widgets/dxFil
 
-    <!--JavaScript-->
-    $("#file-manager").dxFileManager({
-        fileSystemProvider: new DevExpress.fileManagement.CustomFileSystemProvider({
-            // your code
-        }),        
-    });
-
----
-
-Use the [getItems](/api-reference/10%20UI%20Widgets/dxFileManager/5%20File%20System%20Providers/Custom/1%20Configuration/getItems.md '/Documentation/ApiReference/UI_Widgets/dxFileManager/File_System_Providers/Custom/Configuration/#getItems') option to get file system items. The **[fieldName]Expr** options specify the attribute names that store file system item keys, names, sizes, modification dates, and etc.
+3. Implement APIs to handle file operations (add, delete, rename, and etc).
 
 ---
 
 ##### jQuery
 
     <!--JavaScript-->
     $("#file-manager").dxFileManager({
+        // Assigns the Custom file system provider to the widget
         fileSystemProvider: new DevExpress.fileManagement.CustomFileSystemProvider({
-            getItems: function (pathInfo) {
-                // your code
-            }         
+            // Function to get file system items
+            getItems: getItems,
+            // Functions to handle file operations
+            createDirectory: createDirectory,
+            deleteItem: deleteItem
+            // ...
         }),        
     });
+    function getItems (pathInfo) {
+        // your code
+    }
+    function createDirectory(parentDirectory, name) {
+        // your code
+    }
+    function deleteItem(item) {
+        // your code
+    }
 
----
+##### Angular
+
+    <!-- tab: app.component.html -->
+    <dx-file-manager id="fileManager" 
+        [fileSystemProvider]="customFileProvider">
+    </dx-file-manager>
+
+    <!-- tab: app.component.ts -->
+    import { Component } from '@angular/core';
+    import CustomFileSystemProvider from 'devextreme/file_management/custom_provider';
+
+    @Component({
+        selector: 'app-root',
+        templateUrl: 'app/app.component.html',
+        styleUrls: ['app/app.component.css']
+    })
+
+    export class AppComponent {
+        customFileProvider: CustomFileSystemProvider;
+
+        constructor() {
+            // Creates a custom file system provider 
+            this.customFileProvider = new CustomFileSystemProvider({
+                // Function to get file system items
+                getItems,
+                // Functions to handle file operations
+                createDirectory,
+                deleteItem                
+            });
+        }
+        function getItems(pathInfo) {
+            // ...
+        }
+        function createDirectory(parentDirectory, name) {
+            // ...
+        } 
+        function deleteItem(item) {
```

Comment:
I'm not sure that this code is valid. Functions cannot be declared straight in the class.